pol-core/bscript/compiler/astbuilder/FunctionResolver.h

```diff
@@ -0,0 +1,47 @@
+#ifndef POLSERVER_FUNCTIONRESOLVER_H
+#define POLSERVER_FUNCTIONRESOLVER_H
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include <EscriptGrammar/EscriptParser.h>
+
+#include "clib/maputil.h"
+
+#include "compiler/file/SourceLocation.h"
+
+namespace Pol::Bscript::Compiler
+{
+class Function;
+class ModuleFunctionDeclaration;
+class Report;
+
+class FunctionResolver
+{
+public:
+  explicit FunctionResolver( Report& );
+
+  const Function* find( const std::string& scoped_name );
+
+  void register_module_function( ModuleFunctionDeclaration* );
+
+private:
+
+  Report& report;
+
+  //// what about this? (ci_map in Clib)
```

Comment:
I find it easier to grasp if I see `std::map` and its predicate directly. Are you planning to add ci_map to Clib?

Reply:
Yeah, that's exactly why I didn't go with this typedef.  I'll remove the comment.